src/test/java/com/amazon/opendistroforelasticsearch/security/TransportUserInjectorIntegTest.java

```diff
@@ -0,0 +1,167 @@
+
+        // 3. with valid backend roles for injected user
+        UserInjectorPlugin.injectedUser = "injectedadmin|injecttest";
+        try (Node node = new PluginAwareNode(false, tcSettings, Netty4Plugin.class,
+                OpenDistroSecurityPlugin.class, UserInjectorPlugin.class).start()) {
+            waitForInit(node.client());
+            CreateIndexResponse cir = node.client().admin().indices().create(new CreateIndexRequest("captain-logs-2")).actionGet();
+            Assert.assertTrue(cir.isAcknowledged());
+        }
+    }
+
+    @Test
+    public void testOpendistroSecurityUserInjectionWithConfigDisabled() throws Exception {
+        final Settings clusterNodeSettings = Settings.builder()
+                .put(ConfigConstants.OPENDISTRO_SECURITY_UNSUPPORTED_INJECT_USER_ENABLED, false)
+                .build();
+        setup(clusterNodeSettings, new DynamicSecurityConfig().setSecurityRolesMapping("roles_transport_inject_user.yml"), Settings.EMPTY);
+        final Settings tcSettings = Settings.builder()
+                .put(minimumSecuritySettings(Settings.EMPTY).get(0))
+                .put("cluster.name", clusterInfo.clustername)
+                .put("node.data", false)
+                .put("node.master", false)
+                .put("node.ingest", false)
+                .put("path.data", "./target/data/" + clusterInfo.clustername + "/cert/data")
+                .put("path.logs", "./target/data/" + clusterInfo.clustername + "/cert/logs")
+                .put("path.home", "./target")
+                .put("node.name", "testclient")
+                .put("discovery.initial_state_timeout", "8s")
+                .put("opendistro_security.allow_default_init_securityindex", "true")
+                .put(ConfigConstants.OPENDISTRO_SECURITY_UNSUPPORTED_INJECT_USER_ENABLED, false)
+                .putList("discovery.zen.ping.unicast.hosts", clusterInfo.nodeHost + ":" + clusterInfo.nodePort)
+                .build();
+
+        // 1. without user injection
+        try (Node node = new PluginAwareNode(false, tcSettings, Netty4Plugin.class,
+                OpenDistroSecurityPlugin.class, UserInjectorPlugin.class).start()) {
+            waitForInit(node.client());
+            CreateIndexResponse cir = node.client().admin().indices().create(new CreateIndexRequest("captain-logs-1")).actionGet();
+            Assert.assertTrue(cir.isAcknowledged());
+        }
+        
+        // with invalid backend roles
+        UserInjectorPlugin.injectedUser = "ttt|kkk";
+        try (Node node = new PluginAwareNode(false, tcSettings, Netty4Plugin.class,
+                OpenDistroSecurityPlugin.class, UserInjectorPlugin.class).start()) {
+            waitForInit(node.client());
+            CreateIndexResponse cir = node.client().admin().indices().create(new CreateIndexRequest("captain-logs-2")).actionGet();
+            Assert.assertTrue(cir.isAcknowledged());
```

Comment:
Can you add comment that the request is expected to succeed as user injection is disabled.